prepare/libreswan.sh

```diff
@@ -0,0 +1,141 @@
+#!/bin/bash
+
+LIBRESWAN_DIR="/opt/ipsec"
+
+
+# Allow only users with root priviledge to run the script.
+if [ $EUID -ne 0 ]; then
+    echo "This script can be run with root priviledge only." >&2
+    exit 1
+fi
+
+# Check for the correct # of arguments - TODO
+
+
+libreswan_gen_connection ()
+{
+	CONNECTION_CFG="$1"
+	MODE="$2"
+
+	echo "conn roadwarrior_psk
+	auto=add
+	authby=secret
+	pfs=no
+	rekey=no
+	left=172.31.70.1
+	leftsubnet=0.0.0.0/0
+	rightaddresspool=172.29.100.2-10
+	right=%any
+	cisco-unity=yes
+	leftxauthserver=yes
+	rightxauthclient=yes
+	leftmodecfgserver=yes
+	rightmodecfgclient=yes
+	modecfgpull=yes
+	xauthby=alwaysok
+	ike-frag=yes
+	ikev2=never" > "$CONNECTION_CFG"
+	if [ "$MODE" = "aggressive" ]; then
+		echo \
+"	rightid=@yolo
+	aggressive=yes" > "$CONNECTION_CFG"
```

Comment:
The `>` doesn't seem right. It overwrites the entire file.

also, [here-documents](http://tldp.org/LDP/abs/html/here-docs.html) would be a bit simpler to read.